consensus/reactor.go

```diff
@@ -425,14 +461,21 @@ OUTER_LOOP:
 
 		// Send proposal Block parts?
 		if rs.ProposalBlockParts.HasHeader(prs.ProposalBlockPartsHeader) {
-			if index, ok := rs.ProposalBlockParts.BitArray().Sub(prs.ProposalBlockParts.Copy()).PickRandom(); ok {
+			possibleParts := rs.ProposalBlockParts.BitArray().Sub(prs.ProposalBlockParts.Copy())
+			if !possibleParts.IsEmpty() {
+				index := rs.ProposalBlockParts.Abundance.PickRarest(possibleParts, func(setIndex int) {
```

Comment:
why pick func need to accept `setFunc`? can't we call setFunc later if the `Send` succeeds?

Reply:
Yeah, @ebuchman puts a TODO comment in the body of this code wondering what should we do if send fails.